src/proxy/inbound.rs

```diff
@@ -304,8 +311,12 @@ impl Inbound {
 
                 let baggage =
                     parse_baggage_header(req.headers().get_all(BAGGAGE_HEADER)).unwrap_or_default();
+                let src_network_addr = NetworkAddress {
+                    network: conn.dst_network.to_string(), // inbound request source network must match dst network
```

Comment:
ah good catch, copy paste error but yes the idea was to punt on this since the solution isn't trivial. added more detail in https://github.com/istio/ztunnel/pull/486/commits/21c681252f366afc80f444c75be044854a449442 and opened a GH issue in this repo to track potential required updates to HBONE itself to carry the originating network

Reply:
I think we might be able to resolve this by implementing a `check_gateway` function similar to `check_waypoint` in `inbound.rs` and leaving the optional source workload there `None` if we receive from a gateway. looking into this now